cumulus/parachains/runtimes/assets/asset-hub-westend/src/bridge_to_ethereum_config.rs

```diff
@@ -105,19 +123,32 @@ impl snowbridge_pallet_system_frontend::Config for Runtime {
 	type BackendWeightInfo = weights::snowbridge_pallet_system_backend::WeightInfo<Runtime>;
 }
 
-/// `EnsureOriginWithArg` impl for `ForeignTokenCreator` that allows only XCM origins that are
-/// locations containing the class location.
-pub struct ForeignTokenCreator<IsForeign, L = Location>(core::marker::PhantomData<(IsForeign, L)>);
+/// `EnsureOriginWithArg` impl for `ForeignAssetCreatorAsOwner` that
+/// a. allows only XCM origins that are locations containing the class location.
+/// b. check the asset already exists
+/// c. only the owner of the asset can create
+pub struct ForeignAssetCreatorAsOwner<
```

Comment:
Since we're going to duplicate this into many runtimes, I'd suggest it moving it to our `runtime-common` crate at `bridges/snowbridge/runtime/runtime-common/Cargo.toml`.